lib/sneakers/worker.rb

```diff
@@ -72,31 +72,31 @@ def process_work(delivery_info, metadata, msg, handler)
           end
           res = block_to_call.call(deserialized_msg, delivery_info, metadata, handler)
         end
-      rescue StandardError, ScriptError => ex
+      rescue SignalException, SystemExit
+        # ServerEngine handles these exceptions, so they are not expected to be raised within the worker.
+        # Nevertheless, they are listed here to ensure that they are not caught by the rescue block below.
+        raise
+      rescue Exception => ex
```

Comment:
What about interruption signals? Will they work?

Reply:
@kalys Look at the [previous `rescue` block](https://github.com/jondot/sneakers/pull/441/files#diff-89d631e42c0689fc74b63e84d8605cdcR75). `Interrupt` is a subclass of `SignalException`.